src/crypto-providers/util.ts

```diff
@@ -424,21 +424,26 @@ const getAddressParameters = (
   network: Network,
 ): _AddressParameters | null => {
   const addressType = getAddressType(address)
-  try {
-    switch (addressType) {
-      case AddressTypes.BOOTSTRAP:
-        return _packBootStrapAddress(hwSigningData[0], network)
-      case AddressTypes.BASE:
-        return _packBaseAddress(hwSigningData, network)
-      case AddressTypes.ENTERPRISE:
-        return _packEnterpriseAddress(hwSigningData[0], network)
-      case AddressTypes.REWARD:
-        return _packRewardAddress(hwSigningData, network)
-      default: return null
+  const packAddress = () => {
+    try {
+      switch (addressType) {
+        case AddressTypes.BOOTSTRAP:
+          return _packBootStrapAddress(hwSigningData[0], network)
```

Comment:
I am quite concerned with how the singing data are passed into the `_pack*` functions. It might work as it is, but it just does not make sense to me to just pick a random argument (the first one) and pass it in. Either one should check and report more than 1 argument as error, or name the index 0 by some meaningful name if it actually has a special meaning, but I am afraid it does not in this case. Other times, we pass all `hwSigningData` and then randomly pick the first occurence inside e.g. `_packBaseAddress`. Perhaps it relies on some external validation that the signing data contain exactly what is needed, but I really don't get how one would know in *another* place what is actually needed --- even if it was true, it would duplicate the logic then.

Check, for instance, `prepareOutput` --- it just calls `getAddressParameters` in totally the same way for all the change outputs. So the address params would be just the same for any of the change outputs as far as I can see (`_packBaseAddress` does not match the key hash with the given address afaik). Is that a bug or what have I missed? Do we have a test checking that the address params are assigned correctly if there is more than one base address change output?

Reply:
I rewrote the logic, the function now tries to create parameters matching to the address with all provided signing files.